app/services/concerns/xccdf_report/rules.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module XCCDFReport
+  # Methods related to parsing rules
+  module Rules
+    extend ActiveSupport::Concern
+
+    included do
+      def rule_ids
+        test_result.rr.keys
+      end
+
+      def rule_objects
+        @rule_objects ||= @benchmark.items.select do |_, v|
+          v.is_a?(OpenSCAP::Xccdf::Rule)
+        end
+        @rule_objects.map { |rule| rule[1] }
+      end
+
+      def rules_already_saved
+        return @rules_already_saved if @rules_already_saved.present?
+
+        @rules_already_saved = Rule.where(ref_id: rule_objects.map(&:id))
+                                   .includes(:profiles)
+      end
+
+      def add_profiles_to_old_rules(rules, new_profiles)
+        rules.each do |rule|
+          new_profiles.each do |profile|
```

Comment:
Anywhere we have an `ActiveRecord::Relation`, we should probably use `find_each`